kafka/writer.go

```diff
@@ -15,38 +15,100 @@
 package kafka
 
 import (
+	"sync"
+
 	"github.com/Shopify/sarama"
+	log "github.com/Sirupsen/logrus"
 	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/prometheus/storage/remote"
 )
 
+const (
+	namespace = "vulcan"
+	subsystem = "kafka"
+)
+
 // Writer represents an object that encapsulates the behavior of a Kafka
 // producter.
 type Writer struct {
-	producer sarama.SyncProducer
+	producer sarama.AsyncProducer
 	topic    string
+
+	done chan struct{}
+	once *sync.Once
+	wg   *sync.WaitGroup
+
+	successes, errors, enqueued int
```

Comment:
yeah they were only used for logging.  This was before we decided on minimal debug logging globally.